app/jobs/run_submissions_job.rb

```diff
@@ -14,6 +14,9 @@ def run_submission(submission)
     tests.each do |test|
       execution = Execution.find_or_initialize_by(submission_id: submission.id, test_id: test.id)
       
+      if execution.persisted?
```

Comment:
Can we make this:
`next if execution.persisted? && execution.updated_at >= test.updated_at`